### PR TITLE
Fix --pretty=expanded with --remap-path-prefix

### LINKED ISSUE #80832: --pretty=expanded crashes with --remap-path-prefix
I tried this code:

```rust
fn main() {}
```
in `t.rs`.

With `rustc +nightly -Zunstable-options --pretty=expanded ./t.rs` I see the expected output:
```
#![feature(prelude_import)]
#![no_std]
#[prelude_import]
use ::std::prelude::v1::*;
#[macro_use]
extern crate std;
fn main() { }
```

However adding `--remap-path-prefix ./=` causes an ICE:
```
thread 'rustc' panicked at 'called `Option::unwrap()` on a `None` value', compiler/rustc_driver/src/pretty.rs:367:69
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.51.0-nightly (80184183b 2021-01-03) running on x86_64-unknown-linux-gnu

note: compiler flags: -Z unstable-options

query stack during panic:
end of query stack
```

### Meta

`rustc --version --verbose`:
```
binary: rustc
commit-hash: 80184183ba0a53aa4f491753de9502acd3d6920c
commit-date: 2021-01-03
host: x86_64-unknown-linux-gnu
release: 1.51.0-nightly
```

<details><summary>Backtrace</summary>
<p>

```
   0: rust_begin_unwind
             at /rustc/80184183ba0a53aa4f491753de9502acd3d6920c/library/std/src/panicking.rs:493:5
   1: core::panicking::panic_fmt
             at /rustc/80184183ba0a53aa4f491753de9502acd3d6920c/library/core/src/panicking.rs:92:14
   2: core::panicking::panic
             at /rustc/80184183ba0a53aa4f491753de9502acd3d6920c/library/core/src/panicking.rs:50:5
   3: rustc_driver::pretty::get_source
   4: rustc_driver::pretty::print_after_hir_lowering
   5: rustc_interface::queries::<impl rustc_interface::interface::Compiler>::enter
   6: rustc_span::with_source_map
   7: rustc_interface::interface::create_compiler_and_run
   8: rustc_span::with_session_globals
```

</p>
</details>

### PR DESCRIPTION
**Pull request by jsgf**
Per https://github.com/rust-lang/rust/issues/80832, using
--pretty=expanded and --remap-path-prefix results in an ICE.

This is becasue the session source files table is stored in remapped
form, whereas --pretty-expanded looks up unremapped files. This remaps
the path prefixes before lookup.

~~There don't appear to be any existing tests for --pretty=expanded; I'll look into
adding some.~~ Never mind, found the pretty tests.

Fixes #80832

### CODE REVIEW

File: src/test/pretty/issue-80832.rs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// pretty-mode:expanded
```
Comment (davidtwco) [inline src/test/pretty/issue-80832.rs:1]: Could you rename the test to something more descriptive? e.g. `pretty-expanded-with-remapped-path-issue-80832.rs`, there's interest in avoiding new tests just named `issue-XXXXX.rs`.

**Review by davidtwco — CHANGES_REQUESTED** [review]
LGTM, r=me after minor change

File: src/test/pretty/issue-80832.rs (hunk 1 above)
Comment (jsgf) [reply src/test/pretty/issue-80832.rs:1]: Sure, though I'd got comments going the other way in previous PRs.

Comment (jsgf) [reply src/test/pretty/issue-80832.rs]: OK done

**Review by davidtwco — APPROVED** [review]

**Comment by davidtwco** [comment]
Thanks!

@bors r+

**Pull request merged**